packages/x-components/src/x-modules/history-queries/store/actions/update-history-queries-with-search-response.action.ts

```diff
@@ -46,20 +49,29 @@ export const updateHistoryQueriesWithSearchResponse: HistoryQueriesXStoreModule[
   };
 
 /**
- * Take filters from the request and push them into a list.
+ * Creates a selected filters list by comparing request filters and response facets.
+ * Uses the 'filter.id' to match and merge the objects in a single one with all the keys.
  *
+ * @param responseFacets - Facets from the response.
  * @param requestFilters - Filters from the request.
  *
  * @returns A list of selected filters in the history query.
  *
  */
-function createHistoryQueriesFiltersList(
+function getHistoryQueriesFiltersList(
+  responseFacets: InternalSearchResponse['facets'],
   requestFilters: InternalSearchResponse['request']['filters']
 ): Filter[] {
-  return requestFilters
-    ? Object.values(requestFilters).reduce((accFilters, filters) => {
-        accFilters.push(...filters);
-        return accFilters;
-      }, [])
-    : [];
+  if (!requestFilters || !responseFacets) {
+    return [];
+  }
+
+  return Object.values(requestFilters).flatMap(filters =>
+    filters.map(requestFilter => ({
+      ...requestFilter,
+      ...responseFacets
+        .flatMap(facet => facet.filters)
+        .find(filter => filter.id === requestFilter.id)
+    }))
+  );
```

Comment:
You're overriding the fields of the `requestFilter` (where you have the `selected` as `true`) with the response filter value (where you can't know for sure that the filter is correctly selected). You would have to deconstruct the response filter before the `requestFilter`.

In any case, this algorithm can be very slow (you're always iterating all the list of filters from the response) and you are not taking into account possibility that you might not find any matching filter in the response four the request filters.

This is the best I've come up with. I think that it's a good starting point to see if it works well

```suggestion
  return Object.entries(requestFilters).flatMap(([facetId, facetFilters]) => {
    const matchingFacet =
      facetId !== '__unknown__' ? responseFacets.find(facet => facet.id === facetId) : null;

    return facetFilters.reduce<Filter[]>((accFilters, requestFilter) => {
      const matchingFilter = matchingFacet
        ? matchingFacet.filters.find(filter => filter.id === requestFilter.id)
        : responseFacets
            .flatMap(facet => facet.filters)
            .find(filter => filter.id === requestFilter.id);

      if (matchingFilter) {
        accFilters.push({ ...matchingFilter, ...requestFilter });
      }

      return accFilters;
    }, []);
  });
```